doozer/doozerlib/backend/rebaser.py

```diff
@@ -1826,30 +1830,16 @@ def _map_image_name(name, image_map):
             # Re-write the CSV content.
             Path(csv_file).write_text(yaml.dump(csv_obj))
 
-    def _find_previous_versions(self, metadata: ImageMetadata, pattern_suffix='') -> Set[str]:
+    async def _find_previous_versions(self, metadata: ImageMetadata, pattern_suffix='') -> Set[str]:
         """
         Returns: Searches brew for builds of this operator in order and processes them into a set of versions.
         These version may or may not have shipped.
         """
-        # FIXME: This is based on Brew/OSBS. Needs to figure out how to do this with Konflux
-        with self._runtime.pooled_koji_client_session() as koji_api:
-            component_name = metadata.get_component_name()
-            package_info = koji_api.getPackage(component_name)  # e.g. {'id': 66873, 'name': 'atomic-openshift-descheduler-container'}
-            if not package_info:
-                raise IOError(f'No brew package is defined for {component_name}')
-            package_id = package_info['id']  # we could just constrain package name using pattern glob, but providing package ID # should be a much more efficient DB query.
-            pattern_prefix = f'{component_name}-v{metadata.branch_major_minor()}.'
-            builds = koji_api.listBuilds(packageID=package_id,
-                                         state=BuildStates.COMPLETE.value,
-                                         pattern=f'{pattern_prefix}{pattern_suffix}*')
-            nvrs: Set[str] = set([build['nvr'] for build in builds])
-            # NVRS should now be a set including entries like 'cluster-nfd-operator-container-v4.10.0-202211280957.p0.ga42b581.assembly.stream'
-            # We need to convert these into versions like "4.11.0-202205250107"
-            versions: Set[str] = set()
-            for nvr in nvrs:
-                without_component = nvr[len(f'{component_name}-v'):]  # e.g. "4.10.0-202211280957.p0.ga42b581.assembly.stream"
-                version_components = without_component.split('.')[0:3]  # e.g. ['4', '10', '0-202211280957']
-                version = '.'.join(version_components)
-                versions.add(version)
-
-            return versions
+        builds = [build async for build in self._runtime.konflux_db.search_builds_by_fields(
+            where={
+                'name': metadata.get_component_name(),
```

Comment:
if i'm not wrong, "name" is always recorded as meta.distgit_key, so let's use that? Also the reason of restoring "-container", which would mean component and disgit_key, which right now are the same, would be different in future.

Reply:
Done with 1329a5745ab4be52096578d638a16b287280a5a9